knotx-core/src/main/java/io/knotx/server/KnotxServerVerticle.java

```diff
@@ -55,95 +56,73 @@ public void init(Vertx vertx, Context context) {
-        .setHeaderName(csrfConfig.getHeaderName())
-        .setTimeout(csrfConfig.getTimeout());
-
-    Router router = Router.router(vertx);
-    if (options.getAccessLog().isEnabled()) {
-      router.route().handler(LoggerHandler.create(options.getAccessLog().isImmediate(),
-          options.getAccessLog().getFormat()));
-    }
-    router.route().handler(KnotxHeaderHandler.create(options));
-    router.route().handler(SupportedMethodsAndPathsHandler.create(options));
-    router.route().handler(CookieHandler.create());
-    router.route().handler(BodyHandler.create(options.getFileUploadDirectory())
-        .setBodyLimit(options.getFileUploadLimit()));
-
-    router.route().handler(KnotxContextHandler.create());
-
-    options.getDefaultFlow().getRouting().forEach((key, value) -> {
-      value.getItems().forEach(
-          criteria -> {
-            HttpMethod method = HttpMethod.valueOf(key.toUpperCase());
-            if (criteria.isCsrf()) {
-              router.route().method(method)
-                  .pathRegex(criteria.getPath())
-                  .handler(csrfHandler);
+    LOGGER.info("Open API specification location [{}]",
+        options.getRoutingSpecificationLocation());
+
+    OpenAPI3RouterFactory.rxCreate(vertx, options.getRoutingSpecificationLocation())
+        .doOnSuccess(this::registerHandlers)
+        .map(OpenAPI3RouterFactory::getRouter)
+        .doOnSuccess(this::logRouterRoutes)
+        .flatMap(this::configureHttpServer)
+        .subscribe(
+            ok -> {
+              LOGGER.info("Knot.x HTTP Server started. Listening on port {}",
+                  options.getServerOptions().getPort());
+              fut.complete();
+            },
+            error -> {
+              LOGGER.error("Unable to start Knot.x HTTP Server.", error.getCause());
+              fut.fail(error);
             }
-            router.route()
-                .method(method)
-                .pathRegex(criteria.getPath())
-                .handler(KnotxRepositoryHandler.create(vertx, options));
-
-            router.route()
-                .method(method)
-                .pathRegex(criteria.getPath())
-                .handler(KnotxSplitterHandler.create(vertx, options));
-
-            router.route()
-                .method(method)
-                .pathRegex(criteria.getPath())
-                .handler(KnotxEngineHandler
-                    .create(vertx, options, criteria.getAddress(),
-                        criteria.getOnTransition()));
-
-            router.route()
-                .method(method)
-                .pathRegex(criteria.getPath())
-                .handler(KnotxAssemblerHandler.create(vertx, options));
-          }
-      );
+        );
+  }
+
+  private void registerHandlers(OpenAPI3RouterFactory routerFactory) {
+    List<RoutingHandlerFactory> handlerFactories = loadRoutingHandlerFactories();
+    options.getRoutingOperations().forEach(options -> {
```

Comment:
Lambda param `options` covers `private KnotxServerOptions options` field. Please change it to:
```suggestion
    options.getRoutingOperations().forEach(operations -> {
    
```
And in lines below also change `options` to `operations`.